### PR TITLE
check_snmp: disable multiplier when unused

### PR DESCRIPTION
**Pull request by sni**
- if no multiplier is set, simply return the given string. Otherwise we would strip off the unit.
 - if used, allocate new space to hold the result which might be larger than the initial input

### CODE REVIEW

File: plugins/check_snmp.c [hunk 1]
```diff
@@ -1186,16 +1190,17 @@ multiply (char *str)
 	if (fmtstr != "") {
 		conv = fmtstr;
 	}
+	buf = malloc(100);
```
Comment (jdumalaonITRS) [inline plugins/check_snmp.c:1193]: How about we just have this as a static buffer so we do not deal with the malloc? We would have 100 byte leaks for every line in snmpgets output.

Comment (sni) [reply plugins/check_snmp.c:1193]: sounds good, will do

**Review by jdumalaonITRS — APPROVED** [review]
Looks good. Thanks!

**Comment by RincewindsHat** [comment]
close #1849

**Review by RincewindsHat — CHANGES_REQUESTED** [review]
Aparts from that little not, looks good I guess

**Review by RincewindsHat — APPROVED** [review]

**Comment by waja** [comment]
Hmm ... might it be a good idea to ship a test for this?

**Pull request merged**